pkg/cmd/cli/cmd/expose.go

```diff
@@ -119,8 +119,13 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 			supportsTCP := false
 			for _, port := range svc.Spec.Ports {
 				if port.Protocol == kapi.ProtocolTCP {
-					// Pass service target port as the route port
-					cmd.Flags().Set("port", port.TargetPort.String())
+					if len(port.Name) > 0 {
+						// Pass service port name as the route target port, if it is named
+						cmd.Flags().Set("port", port.Name)
+					} else {
+						// Otherwise, omit the targetPort from the route and let it round-robin the one service port
+						cmd.Flags().Set("omit-target-port", "true")
```

Comment:
Can we decide on using port vs target-port as the name of the parameter (and subsequently the name of the flag for setting the route port)? I think the intention is to use target-port. Then we will not require portString to not be zero in the generator, instead of introducing another field to explicitly pass info.